cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -531,18 +540,75 @@ angular.module(PKG.name + '.commons')
       });
     }
 
+    function getPosition(e) {
+      var posx = 0;
+      var posy = 0;
+
+      if (e.pageX || e.pageY) {
+        posx = e.pageX;
+        posy = e.pageY;
+      } else if (e.clientX || e.clientY) {
+        posx = e.clientX + document.body.scrollLeft + document.documentElement.scrollLeft;
+        posy = e.clientY + document.body.scrollTop + document.documentElement.scrollTop;
+      }
+
+      return {
```

Comment:
Ok, have updated to only show the context menu on selected connections. I did this by adding an event listener on a connection every time it is selected, then remove the event listener when it is unselected.